### PR TITLE
Partial revert to fix local-up-cluster.sh

### LINKED ISSUE #60378: local-up-cluster.sh broken by enhance kubeproxy init flag commit
```
/home/dims/go/src/k8s.io/kubernetes/_output/local/bin/linux/amd64/hyperkube flag redefined: config
panic: /home/dims/go/src/k8s.io/kubernetes/_output/local/bin/linux/amd64/hyperkube flag redefined: config

goroutine 1 [running]:
k8s.io/kubernetes/vendor/github.com/spf13/pflag.(*FlagSet).AddFlag(0xc4200b6000, 0xc420bc8dc0)
	/home/dims/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/flag.go:817 +0x9d2
k8s.io/kubernetes/vendor/github.com/spf13/pflag.(*FlagSet).VarPF(0xc4200b6000, 0xaf96b60, 0xc420c4ebb0, 0x4bf8e67, 0x6, 0x0, 0x0, 0x4c7f9b5, 0x23, 0x74a62f)
	/home/dims/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/flag.go:800 +0x120
k8s.io/kubernetes/vendor/github.com/spf13/pflag.(*FlagSet).VarP(0xc4200b6000, 0xaf96b60, 0xc420c4ebb0, 0x4bf8e67, 0x6, 0x0, 0x0, 0x4c7f9b5, 0x23)
	/home/dims/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/flag.go:806 +0x8e
k8s.io/kubernetes/vendor/github.com/spf13/pflag.(*FlagSet).StringVar(0xc4200b6000, 0xc420c4ebb0, 0x4bf8e67, 0x6, 0x0, 0x0, 0x4c7f9b5, 0x23)
	/home/dims/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/string.go:37 +0xa3
k8s.io/kubernetes/cmd/kube-proxy/app.(*Options).AddFlags(0xc420c4ebb0, 0xc4200b6000)
	/home/dims/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:122 +0x74
k8s.io/kubernetes/cmd/kube-proxy/app.NewProxyCommand(0x1)
	/home/dims/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:357 +0x250
main.NewHyperKubeCommand.func3(0xc4201a9440)
	/home/dims/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/hyperkube/main.go:102 +0x26
main.commandFor(0x7ffc9f1b3dec, 0x9, 0xc42052d680, 0xc4202f4380, 0x7, 0x7, 0x8ba6520)
	/home/dims/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/hyperkube/main.go:71 +0xdf
main.main()
	/home/dims/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/hyperkube/main.go:63 +0x230
```

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:
https://github.com/kubernetes/kubernetes/commit/25e110dffc41a5b535adc0426947fce231040f8e broke hyperkube and local-up-cluster.sh. We should revert just the changes in proxy.go to get back to a working hyperkube

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #60378

**Special notes for your reviewer**:
/assign @thockin 
/assign @stewart-yu 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
hmm, this is not enough

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -354,7 +354,7 @@ with the apiserver API to configure the proxy.`,
 		glog.Fatalf("unable to create flag defaults: %v", err)
 	}
 
-	opts.AddFlags(pflag.CommandLine)
+	opts.AddFlags(cmd.Flags())
```
Comment (liggitt) [inline cmd/kube-proxy/app/server.go]: Agree with this change. We shouldn't be registering into global flagsets

**Comment by thockin** [comment]
I am not sure I understand the TODO, but it this fixes a regression, go.

/lgtm
/approve

**Comment by dims** [comment]
@thockin thanks!

**Pull request merged**